test/thumbnailer_test.cc

```diff
@@ -186,6 +187,48 @@ TEST(ThumbnailerTest, NoisyImageTransparent) {
   EXPECT_GT(webp_data->size, 0);
 }
 
+TEST(ThumbnailerPSNRTest, NoisyImageSolid) {
+  libwebp::Thumbnailer thumbnailer = libwebp::Thumbnailer();
+
+  const int pic_count = 10;
+  const uint8_t transparency = 0xff;
+  const bool use_randomized = true;
+  auto test_pics = WebPSamplesGenerator(pic_count, transparency, use_randomized)
+                       .GeneratePics();
+
+  for (int i = 0; i < pic_count; ++i) {
+    thumbnailer.AddFrame(*test_pics[i].get(), i * 500);
+  }
+  std::unique_ptr<WebPData, void (*)(WebPData*)> webp_data(new WebPData,
+                                                           WebPDataDelete);
+  WebPDataInit(webp_data.get());
+  thumbnailer.GenerateAnimationEqualPSNR(webp_data.get());
```

Comment:
It would be nice to check that the psnrs are actually more or less equal. But I haven't checked if there's a public function to easily compute the PSNR. @y-guyon can probably help.
Also there's a lot of repetition in all the tests. I think some repetition in tests is generally ok (it's often more readable) but this seems like a lot. Maybe you can factor out some of the code.

Reply:
There are WebPPlaneDistortion() and WebPPictureDistortion() at your disposal in [encode.h](https://github.com/webmproject/libwebp/blob/master/src/webp/encode.h#L401).
You can either use the latter if you have a WebPPicture nearby, or call the former for each channel otherwise.

Reply:
I think it might be `WebPPictureDistortion`.